app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/attachments/AdditionalOptionButton.kt

```diff
@@ -19,13 +19,35 @@ package com.wire.android.ui.home.messagecomposer.attachments
 
 import androidx.compose.foundation.layout.Box
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalInspectionMode
 import com.wire.android.R
 import com.wire.android.di.hiltViewModelScoped
 import com.wire.android.ui.common.button.WireButtonState
 import com.wire.android.ui.common.button.WireSecondaryIconButton
 import com.wire.android.util.ui.PreviewMultipleThemes
+import kotlinx.coroutines.delay
+
+/**
+ * Represents an additional option button with controlled interactivity for message composer.
+ *
+ * The button's clickability can be controlled via the `isSelected` parameter. This composable also
+ * internally handles preventing rapid successive clicks using the `enableAgain` variable. This
+ * mechanism is particularly important to ensure that, during keyboard transitions (expanding or
+ * collapsing), unintended or unexpected repetitive button clicks do not occur, preventing the
+ * keyboard from collapsing in an unexpected manner.
+ *
+ * @param isSelected Indicates whether the button is selected or not.
+ * @param onClick The action to be performed when the button is clicked.
+ * @param modifier The optional [Modifier] to be applied to this composable.
+ */
+
+private const val BUTTON_CLICK_DELAY_MILLIS = 500L
```

Comment:
question: does it have to be that big? Half a second is already noticeable and I'm afraid that if we have to add such a long delay then maybe we are doing something wrong

Reply:
Lowered to 400ms